zypp/ProblemSolution.cc

```diff
@@ -59,6 +60,17 @@ namespace zypp
     std::string		_details;
     SolutionActionList	_actions;
 
+    template <typename Predicate>
+    bool allActionsMatch( Predicate && predicate ) const
+    {
+      if ( _actions.empty() )
+        return false;
+      for ( const auto & aptr : _actions )
+        if ( not std::invoke( std::forward<Predicate>(predicate), aptr ) )
```

Comment:
Could perfect forwarding the predicate in a loop become a issue too?

Reply:
You mean std::invoke could somehow consume an _rvalue predicate_? I would not assume this.

Reply:
I was thinking about the possibility of a functor consuming itself, but that would also break if you pass that one by reference then...